src/middlewared/middlewared/plugins/jail.py

```diff
@@ -473,3 +475,74 @@ def _import(self, job, jail):
         IOCImage().import_jail(jail)
 
         return True
+
+    @accepts()
+    def get_plugin_versions(self):
+        """
+        Fetches a list of pkg's from the http://pkg.cdn.trueos.org/iocage/
+        repo and returns a dictionary for each plugin
+        """
+        r_pkgs = urllib.request.urlopen('http://pkg.cdn.trueos.org/iocage/All')
+        r_plugins = json.load(
+            urllib.request.urlopen(
+                'https://raw.githubusercontent.com/freenas/'
+                'iocage-ix-plugins/master/INDEX')
+        )
+        pkgs = {
+            'bruserver': ('N/A', '1'),
+            'sickrage': ('Git Branch - master', '1')
+        }
+        pkg_dict = {}
+        plugin_pkgs = {}
+        plugin_manifests = []
+        plugin_translation = {
+            'rslsync': 'btsync',
+            'emby-server': 'emby',
+            'transmission-daemon': 'transmission',
+            'backuppc4': 'backuppc',
+            'nextcloud': 'nextcloud-php56',
+            'tt-rss': 'ttrss',
+            'deluge-cli': 'deluge',
+            'madsonic-standalone': 'madsonic',
+            'subsonic-standalone': 'subsonic',
+            'quassel-core': 'quasselcore',
+            'linux-crashplan': 'crashplan'
+        }
+
+        for i in r_pkgs.readlines():
+            i = i.decode().split('"')
+
+            try:
+                pkg, version = i[1].rsplit('-', 1)
+                pkg_dict[pkg] = version
+            except (ValueError, IndexError):
+                continue  # It's not a pkg
+
+        for plugin, p_dict in r_plugins.items():
+            plugin_manifests.append(p_dict['MANIFEST'])
+
+        for plugin_pkg in plugin_manifests:
+            plugin = plugin_pkg.rsplit('.json')[0]
+            manifest_url = 'https://raw.githubusercontent.com/freenas/'\
+                           f'iocage-ix-plugins/master/{plugin_pkg}'
+            r = json.load(urllib.request.urlopen(manifest_url))
```

Comment:
You're making a new connection to github on every iteration, that is super slow.

There are ways to reuse the same connection, just getting different URLs on same host.

Reply:
Alternatively we can make those in a thread pool.